homeassistant/components/palazzetti/climate.py

```diff
@@ -0,0 +1,155 @@
+"""Support for Palazzetti climates."""
+
+from typing import Any, cast
+
+from pypalazzetti.exceptions import CommunicationError, ValidationError
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_MAC, CONF_NAME, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady, ServiceValidationError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PalazzettiConfigEntry
+from .const import (
+    ACTION_NOT_UNAVAILABLE,
+    DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_MODES,
+    FAN_SILENT,
+    PALAZZETTI,
+)
+from .coordinator import PalazzettiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: PalazzettiConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Palazzetti climates based on a config entry."""
+    coordinator: PalazzettiDataUpdateCoordinator = entry.runtime_data
+    entities: list[PalazzettiClimateEntity] = []
+    if coordinator.client.connected:
+        entities.append(
+            PalazzettiClimateEntity(
+                coordinator=coordinator,
+            )
+        )
+        async_add_entities(entities)
+    else:
+        raise ConfigEntryNotReady
```

Comment:
is this really necessary? wouldn't we raise during the connection if devices are unavailable? If we can add the devices add them, they will show `unavailble` anyways